core/rfb.js

```diff
@@ -1325,18 +1364,144 @@ export default class RFB extends EventTargetMixin {
         Log.Debug("ServerCutText");
 
         if (this._sock.rQwait("ServerCutText header", 7, 1)) { return false; }
+
         this._sock.rQskipBytes(3);  // Padding
-        const length = this._sock.rQshift32();
-        if (this._sock.rQwait("ServerCutText", length, 8)) { return false; }
 
-        const text = this._sock.rQshiftStr(length);
+        let length = this._sock.rQshift32();
+        length = toSigned32bit(length);
+
+        if (this._sock.rQwait("ServerCutText content", Math.abs(length), 8)) { return false; }
+        if (this._viewOnly) {
+            this._sock.rQshiftBytes(Math.abs(length));
+            return true;
+        }
 
-        if (this._viewOnly) { return true; }
+        if (length >= 0) {
+            //Standard msg
+            const text = this._sock.rQshiftStr(length);
 
-        this.dispatchEvent(new CustomEvent(
-            "clipboard",
-            { detail: { text: text } }));
+            this.dispatchEvent(new CustomEvent(
+                "clipboard",
+                { detail: { text: text } }));
+
+        } else {
+            //Extended msg.
+            length = Math.abs(length);
+            const flags = this._sock.rQshift32();
+            let formats = flags & 0x0000FFFF;     //0-15
+            let actions = flags & 0xFF000000; //24-31
```

Comment:
Some weird spacing of the comments here. And perhaps drop the comments? People should be able to deduce the bits easily from those masks.